crates/node-file-trace/src/lib.rs

```diff
@@ -497,8 +459,15 @@ async fn run<B: Backend + 'static, F: Future<Output = ()>>(
         let dir = dir.clone();
         let args = args.clone();
         let sender = sender.clone();
+        let module_options = module_options.clone();
+        let resolve_options = resolve_options.clone();
         Box::pin(async move {
-            let output = main_operation(TransientValue::new(dir.clone()), args.clone().into());
+            let output = main_operation(
+                TransientValue::new(dir.clone()),
+                args.clone().into(),
+                TransientInstance::new(module_options.unwrap_or_default()),
```

Comment:
We still need to hoist this `TransientInstance::new(module_options.unwrap_or_default())` outside the `tt.spawn_root_task`, that way all root tasks can share the transient wrapper around the same `module_options` instance. The current way will create unequal TransientInstances for every task, and kill the caching.